services/slam/slam.go

```diff
@@ -63,6 +64,11 @@ func Named(name string) resource.Name {
 	return resource.NameFromSubtype(Subtype, name)
 }
 
+// FromRobot is a helper for getting the named SLAM service from the given Robot.
+func FromRobot(r robot.Robot, name string) (Service, error) {
+	return robot.ResourceFromRobot[Service](r, Named(name))
```

Comment:
[q] Is `robot.ResourceFromRobot[Service]` guaranteed to always return a 2-argument function that returns a service and an error?

Reply:
Yes this uses go generics and is therefore the contract imposed by the `ResourceFromRobot` function.  It is invoked throughout the codebase in this manner